.github/workflows/Image.yaml

```diff
@@ -202,17 +203,83 @@ jobs:
           path: ${{ env.OCI_ARCHIVE_NAME }}
           key: ${{ github.run_id }}-${{ env.OCI_ARCHIVE_NAME }}
 
-  upload:
+  test:
     runs-on: ubuntu-22.04
     needs: [prepare-build, run-build]
+    name: Test
+    strategy:
+      fail-fast: true
+      matrix: ${{ fromJSON(needs.prepare-build.outputs.build-matrix) }}
+    env:
+      OCI_ARCHIVE_NAME: ${{ matrix.name }}_${{ matrix.commit }}_${{ matrix.revision }}
+    steps:
+      - name: Run tests
+        # Using this actions cause others can have this problem:
+        # https://github.com/convictional/trigger-workflow-and-wait/issues/61
+        uses: mathze/workflow-dispatch-action@v1.2.0
+        id: run-tests
+        env:
+          IS_A_ROCK: ${{ matrix.dockerfile-build == '' && true || false }}
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ github.ref_name }}
+          fail-on-error: true
+          workflow-name: Tests.yaml
+          payload: '{ "oci-image-name": "${{ env.OCI_ARCHIVE_NAME }}", "oci-image-path": "oci/${{ matrix.name }}", "is-a-rock": ${{ env.IS_A_ROCK }}, "cache-key": "${{ github.run_id }}-${{ env.OCI_ARCHIVE_NAME }}", "vulnerability-report-suffix": "${{ env.VULNERABILITY_REPORT_SUFFIX }}" }'
+          trigger-timeout: "5m"
+          wait-timeout: "45m"
+          run-id: dummy
+          use-marker-step: true
+      
+      - name: Write step summary
+        run: |
+          url='${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ steps.run-tests.outputs.run-id }}'
+          echo " - Triggered tests for '${{ matrix.name }}' at [${url}](${url})" >> "$GITHUB_STEP_SUMMARY"
+
+      - name: Enforce test conclusion
+        if: ${{ steps.run-tests.outputs.run-conclusion != 'success' }}
+        # The previous step doesn't always raise an error
+        run: |
+          url='${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ steps.run-tests.outputs.run-id }}'
+          echo "Testing of image '${{ matrix.oci-image }}' failed at [${url}](${url})."
+          exit 1
+
+      # Wait to make sure the test artifacts are uploaded
+      - id: wait-for-artifacts
```

Comment:
are you sure sleep 5 is a good pratice for this kind of things ? Can't we get the id of the test action and check the result ?

Reply:
we have the ID of the run, but that's not the problem. In fact, in the previous step, we even wait for the workflow to finish. However, I've noticed in past runs that GitHub might conclude a run before the artefacts are fully available.

And we don't have an artefact ID to wait for, so this is the ugliest but simplest solution